### PR TITLE
Tools to enable notebook validation and pipeline fix

### PR DESCRIPTION
**Pull request by bettinaheim**
I only realized now that tee cause the exit code of the command to not be properly reflected unless pipefail is set...

This PR should facilitate future notebook changes, and enable to finally pull in https://github.com/NVIDIA/cuda-quantum/pull/848.

### CODE REVIEW

File: scripts/validate_container.sh [hunk 1]
```diff
@@ -245,6 +216,38 @@ do
     echo "============================="
 done
 
+echo "============================="
```
Comment (bmhowe23) [inline scripts/validate_container.sh:219]: I can't tell - did anything change in this snippet, or was it just moved?

Comment (bmhowe23) [reply scripts/validate_container.sh:219]: Never mind, I compared using an external diff tool. I see that 🏳️ , ✅ , and  ❌ were added and no other changes were made (other than moving it). Just curious what the reason for moving it was.

Comment (bettinaheim) [reply scripts/validate_container.sh:219]: It was moved and I added output to make it also show up in the checklist in the step summary.

**Review by bmhowe23 — APPROVED** [review]

File: scripts/validate_container.sh (hunk 1 above)
Comment (bettinaheim) [reply scripts/validate_container.sh:219]: Not much - I added the notebook validation below (separate PR), so I moved the Python part down as well.

**Pull request merged**